shell/platform/darwin/macos/framework/Source/FlutterMetalCompositor.mm

```diff
@@ -11,16 +11,20 @@
 namespace flutter {
 
 FlutterMetalCompositor::FlutterMetalCompositor(
-    FlutterViewController* view_controller,
+    FlutterViewProvider* view_provider,
     FlutterPlatformViewController* platform_views_controller,
     id<MTLDevice> mtl_device)
-    : FlutterCompositor(view_controller),
+    : FlutterCompositor(view_provider),
       mtl_device_(mtl_device),
       platform_views_controller_(platform_views_controller) {}
 
 bool FlutterMetalCompositor::CreateBackingStore(const FlutterBackingStoreConfig* config,
                                                 FlutterBackingStore* backing_store_out) {
-  if (!view_controller_) {
+  // TODO(dkwingsmt): This class only supports single-view for now. As more
+  // classes are gradually converted to multi-view, it should get the view ID
+  // from somewhere.
+  FlutterView* view = GetView(kFlutterDefaultViewId);
+  if (!view) {
```

Comment:
When can this happen?

Reply:
When the caller sends a view ID that the engine does not recognize. When the engine is run as a whole, this should not happen.